From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: xucy10 <someonm20141923@outlook.com>
Date: Tue, 11 Aug 2026 21:59:31 +0800
Subject: [PATCH] Fix compilation errors: PaperHooks final, TIME_BETWEEN_TICKS
 final, TickRegionData lastSavedTime, SaveAllCommand ERROR_FAILED getter,
 Language parseTranslations public, ServerTickRateManager
 getRemainingSprintTicks, EntityType skipRaytracningCheck


diff --git a/ca/spottedleaf/moonrise/paper/PaperHooks.java b/ca/spottedleaf/moonrise/paper/PaperHooks.java
index 08e3c75672e95248140eb0e0907cae5fbe54da34..dd17339fe2f9d7660b20eed05b61d6231302aa82 100644
--- a/ca/spottedleaf/moonrise/paper/PaperHooks.java
+++ b/ca/spottedleaf/moonrise/paper/PaperHooks.java
@@ -29,7 +29,7 @@ import net.minecraft.world.phys.AABB;
 import java.util.List;
 import java.util.function.Predicate;
 
-public final class PaperHooks extends BaseChunkSystemHooks implements PlatformHooks {
+public class PaperHooks extends BaseChunkSystemHooks implements PlatformHooks {
 
     private static final org.slf4j.Logger LOGGER = com.mojang.logging.LogUtils.getLogger();
 
diff --git a/io/papermc/paper/threadedregions/RegionizedServer.java.rej b/io/papermc/paper/threadedregions/RegionizedServer.java.rej
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/io/papermc/paper/threadedregions/TickRegionScheduler.java b/io/papermc/paper/threadedregions/TickRegionScheduler.java
index dc8f23697c28573c20f6cc292f7c9d15ae9ad913..b8d14279b1b27001150b21cf0522997e75068500 100644
--- a/io/papermc/paper/threadedregions/TickRegionScheduler.java
+++ b/io/papermc/paper/threadedregions/TickRegionScheduler.java
@@ -41,7 +41,7 @@ public final class TickRegionScheduler {
     }
 
     public static final int TICK_RATE = 20;
-    public static final long TIME_BETWEEN_TICKS = 1_000_000_000L / TICK_RATE; // ns
+    public static long TIME_BETWEEN_TICKS = 1_000_000_000L / TICK_RATE; // ns
     // Folia start - watchdog
     public static final FoliaWatchdogThread WATCHDOG_THREAD = new FoliaWatchdogThread();
     static {
diff --git a/io/papermc/paper/threadedregions/TickRegions.java b/io/papermc/paper/threadedregions/TickRegions.java
index 97c514673a454b752ab86b944fd143725ca8090d..7a433ee0cb3b925edf728d3f3eb9b954f1ae5d0f 100644
--- a/io/papermc/paper/threadedregions/TickRegions.java
+++ b/io/papermc/paper/threadedregions/TickRegions.java
@@ -206,6 +206,7 @@ public final class TickRegions implements ThreadedRegionizer.RegionCallbacks<Tic
         private final RegionStats regionStats;
 
         private final AtomicBoolean hasPackets = new AtomicBoolean(false);
+        public long lastSavedTime = 0L; // Leaves - last saved time
 
         public volatile ca.spottedleaf.leafprofiler.RegionizedProfiler.Handle profiler; // Folia - profiler
         public final me.earthme.luminol.api.TickRegionData tickRegionDataAPI = new me.earthme.luminol.api.impl.TickRegionDataImpl(this); // Luminol - Tickregion API
diff --git a/net/minecraft/commands/arguments/blocks/BlockInput.java.rej b/net/minecraft/commands/arguments/blocks/BlockInput.java.rej
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/net/minecraft/core/Direction.java.rej b/net/minecraft/core/Direction.java.rej
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/net/minecraft/locale/Language.java b/net/minecraft/locale/Language.java
index 47cd33bce908744e64356b585853d6a8ecf82443..25ffb5fa8047f7f9c65bb9d66b9bc1fff1e093cf 100644
--- a/net/minecraft/locale/Language.java
+++ b/net/minecraft/locale/Language.java
@@ -66,7 +66,7 @@ public abstract class Language {
         };
     }
 
-    private static void parseTranslations(final BiConsumer<String, String> output, final String path) {
+    public static void parseTranslations(final BiConsumer<String, String> output, final String path) {
         try (InputStream stream = Language.class.getResourceAsStream(path)) {
             loadFromJson(stream, output);
         } catch (IOException | JsonParseException e) {
diff --git a/net/minecraft/network/chat/SignedMessageValidator.java.rej b/net/minecraft/network/chat/SignedMessageValidator.java.rej
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/net/minecraft/server/ServerTickRateManager.java b/net/minecraft/server/ServerTickRateManager.java
index eeb2f88723b37bb3cada04bf3098dd46f0ed7316..d487bb35d174ceafbb1832f1e1f5f3c337f6928d 100644
--- a/net/minecraft/server/ServerTickRateManager.java
+++ b/net/minecraft/server/ServerTickRateManager.java
@@ -129,6 +129,8 @@ public class ServerTickRateManager extends TickRateManager {
         this.updateStateToClients();
     }
 
+    public long getRemainingSprintTicks() { return this.remainingSprintTicks; }
+
     public void updateJoiningPlayer(final ServerPlayer player) {
         player.connection.send(ClientboundTickingStatePacket.from(this));
         player.connection.send(ClientboundTickingStepPacket.from(this));
diff --git a/net/minecraft/server/commands/SaveAllCommand.java b/net/minecraft/server/commands/SaveAllCommand.java
index 299991905c24a7018f8ac60d0c16e56fa43cd710..fe34c097584a105fef585e7ecce55771cc45c901 100644
--- a/net/minecraft/server/commands/SaveAllCommand.java
+++ b/net/minecraft/server/commands/SaveAllCommand.java
@@ -12,6 +12,8 @@ import net.minecraft.server.MinecraftServer;
 public class SaveAllCommand {
     private static final SimpleCommandExceptionType ERROR_FAILED = new SimpleCommandExceptionType(Component.translatable("commands.save.failed"));
 
+    public static SimpleCommandExceptionType getErrorFailed() { return ERROR_FAILED; }
+
     public static void register(final CommandDispatcher<CommandSourceStack> dispatcher) {
         dispatcher.register(
             Commands.literal("save-all")
diff --git a/net/minecraft/server/commands/TickCommand.java.rej b/net/minecraft/server/commands/TickCommand.java.rej
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/net/minecraft/server/dedicated/DedicatedPlayerList.java.rej b/net/minecraft/server/dedicated/DedicatedPlayerList.java.rej
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/net/minecraft/server/level/ServerPlayer.java.rej b/net/minecraft/server/level/ServerPlayer.java.rej
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/net/minecraft/server/network/ServerGamePacketListenerImpl.java.rej b/net/minecraft/server/network/ServerGamePacketListenerImpl.java.rej
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/net/minecraft/world/entity/EntityType.java b/net/minecraft/world/entity/EntityType.java
index 31619890f1bd4f458b6f9a31fac8cefaafe09e8e..cfabd7f1f81e55bf515cda9b02a183135e5215fd 100644
--- a/net/minecraft/world/entity/EntityType.java
+++ b/net/minecraft/world/entity/EntityType.java
@@ -57,6 +57,7 @@ public class EntityType<T extends Entity> implements EntityTypeTest<Entity, T>,
     private final Holder.Reference<EntityType<?>> builtInRegistryHolder = BuiltInRegistries.ENTITY_TYPE.createIntrusiveHolder(this);
     public static final Codec<EntityType<?>> CODEC = BuiltInRegistries.ENTITY_TYPE.byNameCodec();
     public static final StreamCodec<RegistryFriendlyByteBuf, EntityType<?>> STREAM_CODEC = ByteBufCodecs.registry(Registries.ENTITY_TYPE);
+    public boolean skipRaytracningCheck = false; // Leaves - entityculling skip raytrace check
     private final EntityType.EntityFactory<T> factory;
     private final MobCategory category;
     private final TagKey<Block> immuneTo;
diff --git a/net/minecraft/world/entity/ExperienceOrb.java.rej b/net/minecraft/world/entity/ExperienceOrb.java.rej
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/net/minecraft/world/entity/item/PrimedTnt.java.rej b/net/minecraft/world/entity/item/PrimedTnt.java.rej
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/net/minecraft/world/entity/monster/EnderMan.java.rej b/net/minecraft/world/entity/monster/EnderMan.java.rej
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/net/minecraft/world/entity/player/Player.java.rej b/net/minecraft/world/entity/player/Player.java.rej
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/net/minecraft/world/item/BlockItem.java.rej b/net/minecraft/world/item/BlockItem.java.rej
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/net/minecraft/world/item/ServerItemCooldowns.java.rej b/net/minecraft/world/item/ServerItemCooldowns.java.rej
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/net/minecraft/world/item/crafting/RecipeManager.java.rej b/net/minecraft/world/item/crafting/RecipeManager.java.rej
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/net/minecraft/world/level/Level.java.rej b/net/minecraft/world/level/Level.java.rej
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/net/minecraft/world/level/NaturalSpawner.java.rej b/net/minecraft/world/level/NaturalSpawner.java.rej
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/net/minecraft/world/level/block/CarvedPumpkinBlock.java.rej b/net/minecraft/world/level/block/CarvedPumpkinBlock.java.rej
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/net/minecraft/world/level/block/ChestBlock.java.rej b/net/minecraft/world/level/block/ChestBlock.java.rej
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/net/minecraft/world/level/block/EnderChestBlock.java.rej b/net/minecraft/world/level/block/EnderChestBlock.java.rej
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/net/minecraft/world/level/block/ObserverBlock.java.rej b/net/minecraft/world/level/block/ObserverBlock.java.rej
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/net/minecraft/world/level/block/RedStoneWireBlock.java.rej b/net/minecraft/world/level/block/RedStoneWireBlock.java.rej
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/net/minecraft/world/level/block/ShulkerBoxBlock.java.rej b/net/minecraft/world/level/block/ShulkerBoxBlock.java.rej
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/net/minecraft/world/level/block/TntBlock.java.rej b/net/minecraft/world/level/block/TntBlock.java.rej
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/net/minecraft/world/level/block/entity/HopperBlockEntity.java.rej b/net/minecraft/world/level/block/entity/HopperBlockEntity.java.rej
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/net/minecraft/world/level/block/piston/PistonBaseBlock.java.rej b/net/minecraft/world/level/block/piston/PistonBaseBlock.java.rej
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/net/minecraft/world/level/block/state/BlockBehaviour.java.rej b/net/minecraft/world/level/block/state/BlockBehaviour.java.rej
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/net/minecraft/world/level/chunk/LevelChunk.java.rej b/net/minecraft/world/level/chunk/LevelChunk.java.rej
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/net/minecraft/world/level/levelgen/feature/SpikeFeature.java.rej b/net/minecraft/world/level/levelgen/feature/SpikeFeature.java.rej
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/net/minecraft/world/level/redstone/NeighborUpdater.java.rej b/net/minecraft/world/level/redstone/NeighborUpdater.java.rej
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
